Added paragraph about Line seperator.

StefanLobbenmeier opened the pull request:
I had some trouble running the Unit tests on my machine, until I realised my dumb mistake. As the LF option is very important in this project (Some Tests will fail), I would suggest adding this friendly reminder :)

bkonyi commented on CONTRIBUTING.md:
```diff
@@ -7,6 +7,9 @@ If you're a first time contributor, welcome aboard! Otherwise, welcome back!
 how to write tests and how to prepare to create a pull request.
 
+# Checkout
+The project uses the line seperator LF in tests. In order to run tests locally on Windows, you have to configure Git to use LF on checkout, as described [here](https://help.github.com/en/articles/dealing-with-line-endings).
```
Good catch! I didn't even think about that :)

bkonyi commented on lib/src/models/submission_impl.dart:
```diff
@@ -136,6 +136,14 @@ class Submission extends SubmissionRef
 
+
+  @override
+  Map<String, String> get infoParams {
```
Style-wise, this can be rewritten as:

```dart
Map<String, String> get infoParams => {
  'sort': commentSortTypeToString(commentSort)
};
```

Not a big deal either way.

StefanLobbenmeier commented:
Probably will not work on this on until Monday, but could you help me with the unit tests? I basically need some kind of documentation on the reply package, couldn’t find anything so far. See #135

StefanLobbenmeier commented:
I just realised there might have been some confusion, because I pushed 2 different issues to the same branch. Let me clarify:
- the first one is only the readme, every Unit Test would fail where the expected.out contained new lines, since the code would check for \n but Windows had \r\n in the files. Nothing to rerecord there, everything working as expected after I configured git properly.
- the second commit/issue is #135 , where I changed the request DRAW makes to the server. Here, we could rerecord the 16 failing test cases, but it should be only one line changes each. However, we should also add additional tests for the other orders.

StefanLobbenmeier commented:
The list of failing unit tests is as follows (raw out here: https://regex101.com/r/uUJ6Bk/1)
> 00:00 +73 ~1 -1: lib/submission/refresh-comments-sanity [E]
> 00:00 +75 ~1 -2: lib/submission/hide-unhide [E]
> 00:00 +75 ~1 -3: lib/submission/hide-unhide-multiple [E]
> 00:00 +76 ~1 -4: lib/submission/submission_flair [E]
> 00:01 +139 ~1 -5: lib/user_content_moderation/contest-mode [E]
> 00:01 +139 ~1 -6: lib/user_content_moderation/set-flair [E]
> 00:01 +139 ~1 -7: lib/user_content_moderation/lock-unlock [E]
> 00:01 +139 ~1 -8: lib/user_content_moderation/nsfw-sfw [E]
> 00:01 +139 ~1 -9: lib/user_content_moderation/spoiler [E]
> 00:01 +139 ~2 -10: lib/user_content_moderation/sticky [E]
> 00:01 +139 ~2 -11: lib/user_content_moderation/remove-approve [E]
> 00:01 +139 ~2 -12: lib/user_content_moderation/distinguish [E]
> 00:01 +139 ~2 -13: lib/user_content_moderation/reports [E]
> 00:01 +140 ~2 -14: lib/user_content/reportable [E]
> 00:01 +141 ~2 -15: lib/user_content/saveable [E]
> 00:01 +141 ~2 -16: lib/user_content/submit-editable-delete [E]

StefanLobbenmeier commented:
Ok seems that I figured it out on my own, I will commit the fixed unit Tests soon.

StefanLobbenmeier commented:
It is actually not as easy as I thought, as the recorded .json Files are being reused for other purposes. I will just change my implementation to default to null instead.

StefanLobbenmeier commented:
Ok I just added another Commit worth pulling: extracting comments from a MoreComments used to result in a list, I added code to transform the list into a tree. This different structure results in a different output in the 
`lib/comment/tons_of_comments_test`
but I think the new behaviour is what should have actually been expected. I just do not know how to generate the corrected .out file.

```
Which: is different.
            Expected: ... PER 0\n230WE MUST GO ...
              Actual: ... PER 0\n2301 \tWE MUS ...

```

bkonyi commented:
Sorry for the latency on this PR. I just want to make sure I'm able to get a proper review in before landing. I'll find time this week!

bkonyi commented on lib/src/models/comment_impl.dart:
```diff
@@ -150,10 +149,56 @@ class MoreComments extends RedditBase with RedditBaseInitializedMixin {
+  }
+
+  List<dynamic> _fillCommentsForests(List<dynamic> fullList) {
+    final first = fullList.first;
```
This will throw an exception if `fullList` is empty. I'd remove this check and do it below.

StefanLobbenmeier replied:
Agreed, that makes more sense.

bkonyi commented on lib/src/models/comment_impl.dart:
```diff
@@ -150,10 +149,56 @@ class MoreComments extends RedditBase with RedditBaseInitializedMixin {
+
+  List<dynamic> _fillCommentsForests(List<dynamic> fullList) {
+    final first = fullList.first;
+    if (fullList.length > 0 && first is Comment) {
```
This should be:

```dart
if (fullList.isNotEmpty && (fullList.first is Comment)) {
```

bkonyi replied:
Why are we explicitly checking that the first element is `Comment`? It's not obvious to me.

StefanLobbenmeier replied:
I agree that I should make it more obvious. This is basically checking for MoreComments, which shouldn’t be resolved here. More Comments also signify the end of the layer.

From my understanding, if the list only contains one Comment or one MoreComment, we don’t have to resolve anything, so maybe I should just change to length > 1?

bkonyi replied:
Yeah, I think that makes sense (I haven't looked at the `Comment` code for awhile). Thanks for clearing that up!

bkonyi commented on lib/src/models/comment_impl.dart:
```diff
@@ -150,10 +149,56 @@ class MoreComments extends RedditBase with RedditBaseInitializedMixin {
+      currentIndex = 0;
+      return _fillCommentsForestsRecursively(fullList, first.depth);
+    } else {
+      return fullList;
```
Can you give an example of which case would result in us just returning `fullList`?

StefanLobbenmeier replied:
As mentioned above  
[MoreComments]  

But I guess this should also catch  
[Comment]
